python/bolt-mcp/tests/test_context.py

```
@@ -0,0 +1,197 @@
+        # Simulate the transport routing the client's response to the pending future.
+        session.pending[req_id].set_result({"role": "assistant", "content": {"type": "text", "text": "ok"}})
+        result = await asyncio.wait_for(task, 1.0)
+        assert result["content"]["text"] == "ok"
+
+    asyncio.run(scenario())
+
+
+def test_stream_call_aclose_does_not_hang_on_blocked_tool():
+    """Client disconnect (aclose) must cancel a tool still awaiting a sample reply."""
+
+    async def scenario():
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `scenario`

Add return type annotation: `None`

(ANN202)